src/errors.ts

```diff
@@ -124,7 +123,15 @@ export function createErrorFromFetchResponse(
   res: Response,
   jsonBody: AcrolinxApiError | undefined,
 ): AcrolinxError {
-  if (jsonBody && jsonBody.type) {
+  if (isAIServiceError(jsonBody)) {
+    return new AcrolinxError({
```

Comment:
Jut curious, why this is different then the default, refer line #147?

Reply:
this was due to jsonBody.type is a property that would be present in ai-service error as well; so it would always fall in first condition;
by having the validation of isAIServiceError first, it will check if type is part of the enum:
https://github.com/acrolinx/sdk-js/pull/118/files#diff-6681300f7fba596107a172032da7abcf6089bbb9bc3f9d1a67423f5dea704903R24

However, this is not necessary anymore, since backend team changed the payload from error.type to error.codeId